pkg/tsdb/opentsdb/callresource.go

```diff
@@ -65,3 +68,386 @@ func (s *Service) HandleSuggestQuery(rw http.ResponseWriter, req *http.Request)
+		aggregators = []string{}
+	}
+
+	sort.Strings(aggregators)
+	sortedResponse, err := json.Marshal(aggregators)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to marshal response: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	for name, values := range res.Header {
+		if name == "Content-Encoding" || name == "Content-Length" {
+			continue
+		}
+		for _, value := range values {
+			rw.Header().Add(name, value)
+		}
+	}
+
+	rw.WriteHeader(res.StatusCode)
+	if _, err := rw.Write(sortedResponse); err != nil {
+		logger.Error("Failed to write response", "error", err)
+		return
+	}
+}
+
+func (s *Service) HandleFiltersQuery(rw http.ResponseWriter, req *http.Request) {
+	logger := logger.FromContext(req.Context())
+
+	dsInfo, err := s.getDSInfo(req.Context(), backend.PluginConfigFromContext(req.Context()))
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to get datasource info: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	u, err := url.Parse(dsInfo.URL)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to parse datasource URL: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	u.Path = path.Join(u.Path, "/api/config/filters")
+	httpReq, err := http.NewRequestWithContext(req.Context(), http.MethodGet, u.String(), nil)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to create request: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	res, err := dsInfo.HTTPClient.Do(httpReq)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to execute request: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	defer func() {
+		if err := res.Body.Close(); err != nil {
+			logger.Error("Failed to close response body", "error", err)
+		}
+	}()
+
+	responseBody, err := DecodeResponseBody(res, logger)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to decode response: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	var filters map[string]json.RawMessage
+	if err := json.Unmarshal(responseBody, &filters); err != nil {
+		logger.Warn("Failed to unmarshal filters response, returning empty array", "error", err)
+		filters = make(map[string]json.RawMessage)
+	}
+
+	keys := make([]string, 0, len(filters))
+	for key := range filters {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
+	sortedResponse, err := json.Marshal(keys)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("failed to marshal response: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	for name, values := range res.Header {
+		if name == "Content-Encoding" || name == "Content-Length" {
+			continue
+		}
+		for _, value := range values {
+			rw.Header().Add(name, value)
+		}
+	}
+
+	rw.WriteHeader(res.StatusCode)
+	if _, err := rw.Write(sortedResponse); err != nil {
+		logger.Error("Failed to write response", "error", err)
+		return
+	}
+}
+
+func (s *Service) HandleLookupQuery(rw http.ResponseWriter, req *http.Request) {
```

Comment:
I was trying to test this and could not get any results to show (using quotes around `cpu.usage` returns `404 Not Found` for the request:

<img width="522" height="787" alt="Screenshot 2025-12-15 at 6 47 44 PM" src="https://github.com/user-attachments/assets/3d504ad1-8529-4a54-a044-18ab475fd3cb" />

I can see the resource call going out to the backend:

<img width="454" height="183" alt="Screenshot 2025-12-15 at 6 48 17 PM" src="https://github.com/user-attachments/assets/5680e099-acc5-42e3-b946-76201b6335e3" />

And tags being returned for that specific metric:

<img width="354" height="330" alt="Screenshot 2025-12-15 at 6 48 37 PM" src="https://github.com/user-attachments/assets/e5c0e0ea-7b88-4537-837a-9bc11424fc9b" />

Reply:
this is due to our test datasource having very little data.

when we call `/api/query` the response includes both `tags` and `aggregatedTags`. ([see the docs](https://opentsdb.net/docs/build/html/api_http/query/index.html#example-aggregated-default-response))
for the tags dropdown selector, we include both of these value sets.

however when querying `/api/search/lookup`, we only receive the `tags`. which is why you see no data. 

if we inspect the response received for `cpu.usage`, or even any of our metrics no `tags` are returned, only `aggregatedTags` which are the values you see above. `['app', 'host', 'host_dns', 'datacenter']`.